pulsar-connect/core/src/main/java/org/apache/pulsar/connect/core/Message.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.connect.core;
+
+/**
+ * Pulsar Connect's Message interface. Message encapsulates the
+ * information about a message being read/written from/to a Source/Sink.
+ */
+public interface Message<T> {
```

Comment:
@srkukarni I was think the Message interface should look like the following:

public interface Message<T> {

    static  <V> Message<V> create(String partitionId, Long sequenceId, V data) {
        return new Message<V>() {
            @Override
            public String getPartitionId() {
                return partitionId;
            }

            @Override
            public Long getSequenceId() {
                return sequenceId;
            }

            @Override
            public V getData() {
                return data;
            }
        };
    }
    /**
     * Retrieves the partition information if any of the message
     * @return The partition id where the
     */
    String getPartitionId();

    /**
     * Retrieves the sequence id of the message
     * @return Sequence Id associated with the message
     */
    Long getSequenceId();

    /**
     * Retrieves the actual data of the message
     * @return The message data
     */
    T getData();
}

Reply:
Users can just use the static create method in the interface